### PR TITLE
Create UploadGeoJSONRequest content as an object

### PR DESCRIPTION
**Pull request by VijayanB**
Signed-off-by: Vijayan Balasubramanian <balasvij@amazon.com>

### Description
Create Request content object to deserialize into an object. This will be used by Transport action
to build logic to index request content as document into an index.
 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/geospatial/GeospatialParser.java [hunk 1]
```diff
@@ -7,19 +7,24 @@
...
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.common.xcontent.XContentType;
 
 /**
  * GeospatialParser provides helper methods to parse/extract/transform input to
  * desired formats.
  */
-public class GeospatialParser {
+public final class GeospatialParser {
```
Comment (martin-gaievski) [inline src/main/java/org/opensearch/geospatial/GeospatialParser.java:20]: why we need final here?

File: src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContent.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    public static final ParseField FIELD_INDEX = new ParseField("index", new String[0]);
+    public static final ParseField FIELD_GEOSPATIAL = new ParseField("field", new String[0]);
+    public static final ParseField FIELD_DATA = new ParseField("data", new String[0]);
+    private final String indexName;
+    private final String geospatialFieldName;
+    private final Object data;
+
+    private UploadGeoJSONRequestContent(String indexName, String geospatialFieldName, Object data) {
+        this.indexName = requireNonNull(indexName);
```
Comment (martin-gaievski) [inline src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContent.java:25]: I suggest we call requireNonNull with second param and put some explicit message about what param value shouldn't be null

File: src/test/java/org/opensearch/geospatial/GeospatialObjectBuilder.java [hunk 3]
```diff
@@ -21,6 +25,9 @@
```
Comment (martin-gaievski) [inline src/test/java/org/opensearch/geospatial/GeospatialObjectBuilder.java:25]: Is this a utility class for building data for test cases?

Comment (VijayanB) [reply src/test/java/org/opensearch/geospatial/GeospatialObjectBuilder.java:25]: Yes, thats correct.

File: src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContent.java (hunk 2 above)
Comment (VijayanB) [reply src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContent.java:25]: Ack

File: src/main/java/org/opensearch/geospatial/GeospatialParser.java (hunk 1 above)
Comment (VijayanB) [reply src/main/java/org/opensearch/geospatial/GeospatialParser.java:20]: To make sure no one can override this .

Comment (martin-gaievski) [reply src/main/java/org/opensearch/geospatial/GeospatialParser.java:20]: This statement is to question :D 
I mean is there a reason why it has been added just now, not with initial change?

Comment (VijayanB) [reply src/main/java/org/opensearch/geospatial/GeospatialParser.java:20]: i missed it in my previous commit :D . Do you prefer separate commit or PR?

Comment (martin-gaievski) [reply src/main/java/org/opensearch/geospatial/GeospatialParser.java:20]: nope that's fine, just want to understand the reason, thank you!

File: src/main/java/org/opensearch/geospatial/GeospatialParser.java [hunk 4]
```diff
@@ -30,4 +35,34 @@
...
+    /**
+     * User inputs are usually deserialized into Map. extractValueAsString will help caller to
+     * extract value from the Map and cast it to string with validation.
+     * @param input User input of type Map
+     * @param key property we would like to extract value of
+     * @return null if key doesn't exist, value as String if it exists, throw exception otherwise
+     */
+    public static String extractValueAsString(final Map<String, Object> input, final String key) {
+        Objects.requireNonNull(key);
+        Objects.requireNonNull(input);
```
Comment (martin-gaievski) [inline src/main/java/org/opensearch/geospatial/GeospatialParser.java:48]: can we do the same addition with descriptive user message here as well?

**Review by martin-gaievski — APPROVED** [review]
Lgtm, thank you

File: src/main/java/org/opensearch/geospatial/GeospatialParser.java (hunk 4 above)
Comment (VijayanB) [reply src/main/java/org/opensearch/geospatial/GeospatialParser.java]: Ack

**Review by jmazanec15 — APPROVED** [review]

**Pull request merged**